### DISCUSSION TITLE
How to insert an array of values in a field so that field is generated as array rather than a string [Sep 08]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#6392] [Discord Threads]
> 🙋‍♂️ **sajeesh**  
#### - DB - mariadb  
![image0](https://cdn.discordapp.com/attachments/661905455894888493/1138520705437274152/image.png)  
![image1](https://cdn.discordapp.com/attachments/661905455894888493/1138520705818951880/image.png)

**Comment by o1lab** [comment]
> 🙋‍♂️ **sajeesh**  
#### Moving this to a thread.    
####    
#### I scanned the entire documentation and github issues / discussion but couldn't find a solution.  
#  
> 👨‍🚀 **wingkwong**  
#### json type  
![image0](https://cdn.discordapp.com/attachments/1138520706028687481/1138686396857589810/image.png)  
#  
> 🙋‍♂️ **sajeesh**  
#### sorry didn't get that ... how did you achieve this. Did you write "json" instead of "pros" in the example i gave?  
#  
> 👨‍🚀 **wingkwong**  
#### use JSON column type  
#  
> 👨‍🚀 **wingkwong**  
#### the field name doesnt matter  
#  
> 🙋‍♂️ **sajeesh**  
#### i used the json column type itself. see the example  
![image0](https://cdn.discordapp.com/attachments/1138520706028687481/1138831181098532994/image.png)  
![image1](https://cdn.discordapp.com/attachments/1138520706028687481/1138831181350182952/image.png)  
![image2](https://cdn.discordapp.com/attachments/1138520706028687481/1138831181639594034/image.png)  
#  
> 🙋‍♂️ **sajeesh**  
#### <@495466997304262658>  suggestion?  
#  
> 👨‍🚀 **wingkwong**  
#### when u double click the cell, does the data include line breaks? i.e. formatted  
#  
> 👨‍🚀 **wingkwong**  
#### or just one line  
#  
> 🙋‍♂️ **sajeesh**  
#### i don't put any space, but aftere save it adds spaces automatically.  
#  
> 🙋‍♂️ **sajeesh**  
#### hey just to let you know i just install a fresh mysql db and did the same test and it worked. however it is not working on Mariadb setup. Not sure how is it possible, but that's the truth. Is it something to do with Mariadb ??  
#  
> 🙋‍♂️ **sajeesh**  
#### it didn't work in SQLite as well. So only MYSQL is supporting it seems.  
#  
> 👨‍🚀 **wingkwong**  
#### i was using pg tho  
#  
> 👨‍🚀 **wingkwong**  
#### true. i can see the issue in sqlite  
#  
> 🙋‍♂️ **sajeesh**  
#### i would have loved if it worked in SQLite since my project doesn't require multiple concurrent connections. Unfortunately, at least for me, JSON field is behaving properly in MYSQL.    
####    
#### Any plans to fix this in SQLite?  
#  
> 🙋‍♂️ **sajeesh**  
#### JSON type for SQLite works like charm in both Directus and Pocketbase, but somehow i love using NocoDB  
#  
> 👨‍🚀 **wingkwong**  
#### actually sqlite stores json as text, unlike pg / mysql  
#  
> 👨‍🚀 **wingkwong**  
#### in UI, we parse the value so it looks fine  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1138520706028687481)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
